editor/editor_node.cpp

```diff
@@ -964,6 +964,12 @@ void EditorNode::_resources_changed(const Vector<String> &p_resources) {
 	if (changed.size()) {
 		for (Ref<Resource> &res : changed) {
 			res->reload_from_file();
+
+			// Update exports to update the properties in the inspector.
+			Ref<Script> scr = Object::cast_to<Script>(res.ptr());
+			if (scr.is_valid()) {
+				scr->update_exports();
+			}
```

Comment:
Maybe this should be in `Script::reload_from_file()`.

Reply:
There's already a` reload_from_file` just above and it was not sufficient.

Reply:
Sorry, I meant that `update_exports()` should be added into the Script's `reload_from_file` implementation instead of having it here, since every time we reload the script we also want to update the exports.
For instance `ScriptEditor::_close_discard_current_tab` also uses `reload_from_file` and it should call `update_exports` too.